daemon/logger/splunk/splunk.go

```diff
@@ -44,22 +46,44 @@ type splunkLogger struct {
 	nullMessage *splunkMessage
 }
 
+type splunkLoggerInline struct {
+	splunkLogger
+
+	nullEvent *splunkMessageEvent
+}
+
+type splunkLoggerJSON struct {
+	splunkLoggerInline
+}
+
+type splunkLoggerRaw struct {
+	splunkLogger
+
+	prefix []byte
+}
+
 type splunkMessage struct {
-	Event      splunkMessageEvent `json:"event"`
-	Time       string             `json:"time"`
-	Host       string             `json:"host"`
-	Source     string             `json:"source,omitempty"`
-	SourceType string             `json:"sourcetype,omitempty"`
-	Index      string             `json:"index,omitempty"`
+	Event      interface{} `json:"event"`
```

Comment:
Because we have 3 different formats of the Events now - that fields can be a simple string or `splunkMessageEvent`. The same in `splunkMessageEvent.Line`, it can be a simple string or JSON struct (map of interfaces)